### PR TITLE
Deprecate this crate

### PR DESCRIPTION
**Pull request by bnjbvr**
We are deprecating this crate, since regalloc2 is more modern, maintained and expected to be faster both in compile times and have better generated code quality.

@cfallin If you agree with this, once we've merged this, I'll archive this repository and disable pull requests/issues. If there's any interesting issues that also makes sense for regalloc2, we can re-open them in regalloc2's issue tracker.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,6 +1,10 @@
 regalloc.rs
 ===
 
+⚠ This project has been deprecated in favor of the [regalloc2](https://github.com/cfallin/regalloc2), which is maintained and expected to be faster. No security updates will be made, and contributions will be disabled.
```
Comment (bjorn3) [inline README.md:4]: Is regalloc2 not under the bytecodealliance organization?

Comment (bnjbvr) [reply README.md:4]: Aouch, good catch, thanks!

Comment (bnjbvr) [inline README.md:4]:
```suggestion
⚠ This project has been deprecated in favor of the [regalloc2](https://github.com/bytecodealliance/regalloc2), which is maintained and expected to be faster. No security updates will be made, and contributions will be disabled.
```

**Review by cfallin — APPROVED** [review]
Thanks @bnjbvr -- and thanks to regalloc.rs for carrying Cranelift well for several years!

**Comment by cfallin** [comment]
cargo-deny is failing because the crate `unicode-ident` (pulled in via some proc-macro helpers, then serde) added an "and Unicode license" qualifier to its license field. Since this crate is no longer used in downstream contexts that have strict license requirements, and because such a license would likely not affect the actual compiled code, and because the tree today is the same as it was yesterday (this PR doesn't introduce any changes to dependencies), I'm going to go ahead and merge this against the CI failure.

**Pull request merged**